src/tasks/beacon/beacon_task.c

```diff
@@ -70,8 +72,14 @@ size_t serialize_slate(slate_t *slate, uint8_t *data)
                           .solar_current = slate->solar_current,
                           .device_status = get_device_status(slate)};
 
-    memcpy(data + name_len + 1, &stats, sizeof(beacon_stats));
-    return name_len + 2 + sizeof(beacon_stats);
+    // 2 Extra bytes: 1 for length of string, 1 for \0 terminator
+    memcpy(data + name_len + 2, &stats, sizeof(beacon_stats));
```

Comment:
this was a bug I introduced, it should still be `data + name_len + 1`. The length of string field was removed from beacon packing (we read until \0 for Kaitai decoding).